### ISSUE TITLE
Lingering issues with percentage and stats selector functionality

### ISSUE THREAD

**Issue by patudom** [cosmicds/cosmicds#246]
We are merging in #243 for now, but I'm noting here some issues that still remain with its functionality.

- [x] When you have a stat displayed and then click on a percentage range, the stat marker remains in the top viewer, but disappears in the 2nd viewer.
<img width="1315" alt="Screen Shot 2023-07-15 at 11 54 58 AM" src="https://github.com/cosmicds/cosmicds/assets/12750048/15693a7d-afe1-41a8-a3b0-f9a0772cac26">

- [x] The percentage selector is not handling small class size correctly. I would expect the inner 50% of data for 6 students to include 3 or 4 values. This is displaying 5 values.
<img width="634" alt="Screen Shot 2023-07-15 at 11 28 59 AM" src="https://github.com/cosmicds/cosmicds/assets/12750048/6fc3ad09-8af7-4254-8bc4-ac42274f24a1">

**Comment by patudom** [comment]
- [ ] When a student clicks any ? to open a definition, it would make sense for that stat/range to be displayed if it is not already. Is that something that would be straightforward to add? (i.e., clicking ? activates the radio button if it isn't already active.)

**Comment by johnarban** [comment]
One other issue: The highlighted range in 13-15, but the label says 14-15. It looks like only one thing is selected in this. and it just so happens that the highest value in that bin is 13.832..... (this is using student 4070)

<img width="738" alt="image" src="https://github.com/cosmicds/cosmicds/assets/7862929/5fe5da11-6a4a-4365-bad0-c5bdd141905d">

**Comment by Carifio24** [comment]
There are fixes for these issues in #248. The only thing that I'm not 100% satisfied with is the percentage/figure title - I moved it over to the left a bit to try and avoid clashing with the legend, but I'm still looking for a robust solution.

In that PR, I changed the way that the percentile works. What I had been doing is using the `Data`'s built in percentile calculation to find the upper and lower percentiles, and then defining a range subset between these two values. But when one can have duplicate data values, this isn't the same thing as actually taking X% of the points, which is particularly noticeable when it pops up in smaller data sets. I've modified the percentile calculation to be based on sorting indices, meaning that we're actually taking X% of the data points (or as close to X% as the size of the data will allow).

I also modified the range value that's reported. As we discussed @patudom, we can now set a resolution (in decimal places) that determines the reported value, with one additional piece of logic to make sure that the rounded value lies within the relevant bin (that is, that rounding doesn't move the value over into the next bin).
